### DISCUSSION TITLE
Typescript: Using variations of the generated model type in version 3

### DISCUSSION THREAD

**Discussion by amcjen** [prisma/prisma#9405] [Q&A] [answered]
Following the directions [here](https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/operating-against-partial-structures-of-model-types#problem-using-variations-of-the-generated-model-type) with regards to dealing with TypeScript issues, and it seems that as of version 3.1, `UserArgs` is no longer available (per the screenshot error below)? 

What is the proper way to achieve this in >= 3.1?

![Untitled 2](https://user-images.githubusercontent.com/175249/134537041-a7450492-c839-45dc-8b7b-6b138c4e28ee.jpg)

**Comment by ryands17** [comment, marked as answer]
@amcjen 👋 

This works fine for me. Tested on a simple model using Prisma `3.1.1`:

![image](https://user-images.githubusercontent.com/19697099/134855005-b89163ff-32d2-47e9-86e8-8e319c9331ab.png)

Could you try re-generating the client and check?

**Reply by amcjen** [reply]
Strange. I'm on 3.1.1 as well and I'm still having the problem. I removed and reinstalled Prisma, removed the `node_modules/.prisma` directory and regenerated the client (screenshot below), and still get the same `...has no exported member 'UserArgs'`.

![image](https://user-images.githubusercontent.com/175249/135154747-9d8b1d5a-74a0-4e30-807b-c24865c7a4b1.png)

**Reply by ryands17** [reply]
@amcjen Would you have a small reproduction to share so that I can check?

**Reply by amcjen** [reply]
Ah wow, fired up the quickstart and it worked fine, then realized `UserArgs` is not a constant, but is specific to the model, and refers to the `User` object! Looking at it now, it's obvious because there's no other reference to `User`. So my code just needed the proper `productArgs` change to reference the model, and all is well.

Thanks for the clarification! This is _much_ better than hand-writing the types as we had to resort to doing before getting this working. 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
